autosklearn/ensemble_builder.py

```diff
@@ -202,28 +209,27 @@ def ensemble_builder_process(
                         time_left=time_left_for_ensembles-elapsed_time,
                         iteration=iteration,
                     )
-                )
-                logger.info(
-                    "{} Started Ensemble builder job at {} for iteration {}. Future={}/{}".format(
-                        # Log the client to make sure we
-                        # remain connected to the scheduler
-                        client,
-                        time.strftime("%Y.%m.%d-%H.%M.%S"),
-                        iteration,
-                        futures[-1],
-                        len(futures),
-                    ),
-                )
-        except Exception as e:
-            exception_traceback = traceback.format_exc()
-            error_message = repr(e)
-            logger.critical(exception_traceback)
-            logger.critical(error_message)
+
+                    logger.info(
+                        "{}/{} Started Ensemble builder job at {} for iteration {}.".format(
+                            # Log the client to make sure we
+                            # remain connected to the scheduler
+                            future,
+                            client,
+                            time.strftime("%Y.%m.%d-%H.%M.%S"),
+                            iteration,
+                        ),
+                    )
+                iteration += 1
+            except Exception as e:
+                exception_traceback = traceback.format_exc()
+                error_message = repr(e)
+                logger.critical(exception_traceback)
+                logger.critical(error_message)
 
         # If pynisher kills a run, the result
         # might be None -- so no new timestamp info
-        done_futures = [f for f in futures if f.done()]
-        for future in done_futures:
+        if future.done():
```

Comment:
This looks like it could fail - assume that we're in the 1st iteration and fail in the try/except above. Now, future is still `None`.

Reply:
Thanks, I have fixed that.